contrib/platform/test/com/sun/jna/platform/win32/COM/ShellApplicationWindowsTest.java

```diff
@@ -39,16 +44,41 @@
 
 public class ShellApplicationWindowsTest {
 
+    private static final Guid.CLSID CLSID_InternetExplorer = new Guid.CLSID("{0002DF01-0000-0000-C000-000000000046}");
+
     static {
         ClassLoader.getSystemClassLoader().setDefaultAssertionStatus(true);
     }
 
+    private PointerByReference ieApp;
+    private Dispatch ieDispatch;
+
     @Before
     public void setUp() throws Exception {
-        Ole32.INSTANCE.CoInitializeEx(Pointer.NULL, Ole32.COINIT_MULTITHREADED);
+        WinNT.HRESULT hr;
 
-        // Launch IE in a manner that should ensure it opens even if the system default browser is Chrome, Firefox, or something else.
-        Runtime.getRuntime().exec("cmd /c start iexplore.exe -nohome \"about:blank\"");
+        hr = Ole32.INSTANCE.CoInitializeEx(Pointer.NULL, Ole32.COINIT_MULTITHREADED);
```

Comment:
Will `tearDown()` still run if this fails? Making sure you don't over- `CoUninitialize()`.

Reply:
> Will `tearDown()` still run if this fails? Making sure you don't over- `CoUninitialize()`.

"fails" would only include an `hr` of `RPC_E_CHANGED_MODE`. Both `S_OK` and `S_FALSE` require the uninitialize.